src/JsonReader.cs

```diff
@@ -548,127 +570,162 @@ sealed class NonProperty : IJsonProperty<Unit, JsonReadResult<Unit>>
+                                {
+                                    case { Incomplete: true }:
+                                        return reader.Suspend((sm, values));
+                                    case { Error: { } err }:
+                                        return new JsonReadError(err);
+                                    case { Value: var val }:
+                                        value = (true, val);
+                                        return null;
+                                }
+                            }
+
+                            if (values.CurrentPropertyIndex is { } nextPropertyIndex)
+                            {
+                                var error = nextPropertyIndex switch
+                                {
+                                    1  => ReadPropertyValue(property1,  ref reader, ref values.V1,  ref sm, ref values),
+                                    2  => ReadPropertyValue(property2,  ref reader, ref values.V2,  ref sm, ref values),
+                                    3  => ReadPropertyValue(property3,  ref reader, ref values.V3,  ref sm, ref values),
+                                    4  => ReadPropertyValue(property4,  ref reader, ref values.V4,  ref sm, ref values),
+                                    5  => ReadPropertyValue(property5,  ref reader, ref values.V5,  ref sm, ref values),
+                                    6  => ReadPropertyValue(property6,  ref reader, ref values.V6,  ref sm, ref values),
+                                    7  => ReadPropertyValue(property7,  ref reader, ref values.V7,  ref sm, ref values),
+                                    8  => ReadPropertyValue(property8,  ref reader, ref values.V8,  ref sm, ref values),
+                                    9  => ReadPropertyValue(property9,  ref reader, ref values.V9,  ref sm, ref values),
+                                    10 => ReadPropertyValue(property10, ref reader, ref values.V10, ref sm, ref values),
+                                    11 => ReadPropertyValue(property11, ref reader, ref values.V11, ref sm, ref values),
+                                    12 => ReadPropertyValue(property12, ref reader, ref values.V12, ref sm, ref values),
+                                    13 => ReadPropertyValue(property13, ref reader, ref values.V13, ref sm, ref values),
+                                    14 => ReadPropertyValue(property14, ref reader, ref values.V14, ref sm, ref values),
+                                    15 => ReadPropertyValue(property15, ref reader, ref values.V15, ref sm, ref values),
+                                    16 => ReadPropertyValue(property16, ref reader, ref values.V16, ref sm, ref values),
+                                    _ => throw new InvalidOperationException()
+                                };
+
+                                if (error is not null)
+                                    return error.Value;
```

Comment:
Prefer binding within the pattern:

```suggestion
                                if (error is { } someResult)
                                    return someResult;
```